### DISCUSSION TITLE
ChangeLog API KirimWA.id - v1.0-RC24

### DISCUSSION THREAD

**Discussion by rioastamal** [kirimwa-id/forum#20] [Pengumuman]
# ChangeLog dari API KirimWA.id

## v1.0-RC24 (2025-08-15)

Rilis ke-24 developer preview. Bug fixes yang dihadirkan:

- Bug fix: Tidak bisa mengirim pesan ke groups
- Bug fix: App name di Linked Devices
- Bug fix: Add image/jpeg ke dalam daftar list content-type
- Bug fix: Enterprise wrong calculation
- Bug fix: PDF document dengan caption
- Bug fix: Return error ketika bukan sebuah WhatsApp number
- Bug fix: http rate limit new quota
- Bug fix: batch-messages monthtly quota
- Bug fix: reconnect issue
- Menghapus dukungan untuk buttons dan list message

## v1.0-RC21 (2023-02-20)

Rilis ke-21 developer preview. Fitur-fitur yang dihadirkan:

- Fitur baru: interactive message (list message dan reply button)
- Bug fix untuk manajamen payload size

## v1.0-RC20 (2023-01-04)

Rilis ke-20 developer preview. Fitur-fitur yang dihadirkan:

- Bug fix untuk pesan dengan tipe image dimana caption harusnya tetap kosong jika tidak diisi #157 
- Internal code optimization

## v1.0-RC19 (2022-12-31)

Rilis ke-19 developer preview. Fitur-fitur yang dihadirkan:

- Perbaikan pada internal queue untuk pemanggilan Webhook

## v1.0-RC18 (2022-10-23)

Rilis ke-18 developer preview. Fitur-fitur yang dihadirkan:

- Bug fix untuk session yang memiliki lebih dari satu device

## v1.0-RC17 (2022-10-14)

Rilis ke-17 developer preview. Fitur-fitur yang dihadirkan:

- Internal improvement for logging

## v1.0-RC16 (2022-10-09)

Rilis ke-16 developer preview. Fitur-fitur yang dihadirkan:

- Bug fix session yang tidak stabil

## v1.0-RC15 (2022-07-21)

Rilis ke-15 developer preview. Fitur-fitur yang dihadirkan:

- Bug fix fitur mentions tidak bekerja
- Bug fix device status tidak berubah ketika koneksi telah terbuka
- Bug fix nama atribut pada /qr respon
- Perbaikan pada webhook dengan tidak mengirim tipe pesan yang tidak didukung

## v1.0-RC14 (2022-07-19)

Rilis ke-14 developer preview. Fitur-fitur yang dihadirkan:

- Mendukung WhatsApp multi-device

## v1.0-RC13 (2022-04-03)

Rilis ke-13 developer preview. Fitur-fitur yang dihadirkan:

- Menambah dukungan untuk tipe pesan pada webhook incoming message. Saat ini tipe yang didukung adalah: text, image, document, video, contact, dan location.
- Terdapat filter untuk nomor yang ingin diproses pada incoming message
- API /quotas sekarang menampilkan jumlah quota untuk incoming media bytes per bulan

## v1.0-RC12 (2021-12-12)

Rilis ke-12 developer preview. Fitur-fitur yang dihadirkan:

- API baru /stats untuk melihat statistik umum dari API KirimWA.id
- Optimasi saat aplikasi melakukan cold start
- Penambahan beberapa metrics baru untuk memudahkan analytics (internal)

## v1.0-RC11 (2021-11-21)

Rilis ke-11 developer preview. Fitur-fitur yang dihadirkan:

- Timeout baru ketika melakukan request ke /groups
- Bug fix pada webhook send_message_response (diskusi #57)

## v1.0-RC10 (2021-10-31)

Rilis ke-10 developer preview. Fitur-fitur yang dihadirkan:

- Bug fixes incoming message tidak terproses karena salah deteksi quota

## v1.0-RC9 (2021-10-30)

Rilis ke-9 developer preview. Fitur-fitur yang dihadirkan:

- Kirim pesan terjadwal
- Kirim pesan secara batch (max. 50 pesan per batch)
- Webhook device status berubah (disconnected atau connected)
- Bug fixes pada incoming message

## v1.0-RC8 (2021-10-04)

Rilis ke-8 developer preview. Fitur-fitur yang dihadirkan:

- Kirim pesan document (pdf, docx, xlxs, wav, mp3, dll)
- Mention anggota grup diawali dengan tanda '@' 
- Daftar kontak pada sebuah grup

## v1.0-RC7 (2021-09-17)

Rilis ke-7 developer preview. Fitur-fitur yang dihadirkan:

- Pesan masuk dari nomor sendiri dapat diteruskan ke webhook

## v1.0-RC6 (2021-09-11)

Rilis ke-enam developer preview. Fitur-fitur yang dihadirkan:

- Webhook untuk pesan masuk (incoming message), sementara hanya mendukung pesan teks
- API baru untuk melihat pesan yang telah masuk
  - `GET /incoming-messages`
- API baru untuk melakukan reconnect tanpa perlu scan ulang QR code
  - `POST /reconnect`
- Perbaikan pada webhook call
- Perbaikan pada ping checker untuk tetap membuka koneksi

## v1.0-RC5 (2021-08-25)

Rilis ke-lima developer preview. Fitur-fitur yang dihadirkan:

- Menambahkan cache untuk hasil `GET /groups`
- Perbaikan scheduler untuk pengiriman pesan ke grup
- Mengganti Docker base image dari Alpine ke Debian

## v1.0-RC4 (2021-08-21)

Rilis ke-empat developer preview. Fitur-fitur yang dihadirkan:

- Perbaikan pada scheduler pemrosesan pesan

## v1.0-RC3 (2021-08-16)

Rilis ke-tiga developer preview. Fitur-fitur yang dihadirkan:

- Bug fixes pada endpoint `GET /groups`
- Bug fixes pada internal connection handling

## v1.0-RC2 (2021-08-13)

Rilis ke-dua developer preview. Fitur-fitur yang dihadirkan:

- API untuk Group list `GET /groups`
- Mengirim pesan ke Group lewat endpoint `POST /messages`

## v1.0-RC1 (2021-07-22)

Rilis pertama developer preview. Fitur-fitur yang dihadirkan:

- API untuk devices
  - `GET /devices`
  - `GET /devices/id`
  - `POST /devices`
  - `DELETE /devices`
- API untuk QR
  - `GET /qr`
- API untuk mengirim pesan. Format yang didukung hanya teks dan gambar.
  - `POST /messages`
  - `GET /messages`
  - `GET /messages/id`
- API untuk Quota
  - `GET /quotas`
- Webhooks. Dukungan untuk webhook pesan yang telah dikirim dimana `webhook_type` = `send_message_response`.
  - `POST /webhooks`
  - `GET /webhooks`
  - `DELETE /webhooks/id`
  - `GET /webhook-responses`
  - `GET /webhook-responses/id`

**Comment by rioastamal** [comment]
Halo semua,

Fitur v1.0-RC16 baru saja diluncurkan untuk perbaikan session yang tidak stabil pada multi-device. Efek dari update ini adalah semua session aktif menjadi tidak valid dan harus re-scan ulang /qr code.

**Reply by zonegoh** [reply]
setelah rescan, pesan semua tidak terkirim... response status "fail"

**Reply by mediasaranainformatika** [reply]
Terima kasih mas rio... Pesan sdh bsa terkirim kembali

**Reply by zonegoh** [reply]
apakah support untuk whatsapp business mas?

**Reply by mediasaranainformatika** [reply]
Kebetulan sy pake WA bisnis pak

**Comment by rioastamal** [comment]
### Update v1.0-RC18

Bug fix untuk session yang tidak stabil bagi akun yang memiliki lebih dari satu device. Jika Anda menggunakan multiple device pada API KirimWA.id kemungkinan session anda akan tidak valid dan perlu untuk logout dan rescan QR. Bagi yang hanya memiliki satu device harusnya session tetap valid.
